diff -up ./usbguard-selinux-0.0.3/usbguard.te.cpuinfo ./usbguard-selinux-0.0.3/usbguard.te
--- ./usbguard-selinux-0.0.3/usbguard.te.cpuinfo	2020-06-18 15:53:40.161615146 +0200
+++ ./usbguard-selinux-0.0.3/usbguard.te	2020-06-18 15:54:28.399982328 +0200
@@ -77,6 +77,8 @@ auth_read_passwd(usbguard_t)
 dev_list_sysfs(usbguard_t)
 dev_rw_sysfs(usbguard_t)
 
+kernel_read_system_state(usbguard_t)
+
 list_dirs_pattern(usbguard_t,usbguard_conf_t,usbguard_conf_t)
 read_files_pattern(usbguard_t,usbguard_conf_t,usbguard_conf_t)
 dontaudit usbguard_t usbguard_conf_t:file write;
